README.md

```diff
@@ -1,380 +1,414 @@
+The *umzug* lib is a framework agnostic migration tool for Node.JS. The tool itself is not specifically related to databases but basically provides a clean API for running and rolling back tasks.
 
-## Storages
+* Programmatic API for migrations
+* Database agnostic
+* Supports logging of migration process
+* Supports multiple storages for migration data
+  
+## Documentation
 
-### JSONStorage
-Using the [JSONStorage](src/storages/JSONStorage.js) will create a JSON file which will contain an array with all the executed migrations. You can specify the path to the file. The default for that is `umzug.json` in the working directory of the process.
+### Minimal Example
 
-#### Options
+The following example uses a Sqlite database through sequelize and persists the migration data in the database itself through the sequelize storage.
 
-```js
-{
-  // The path to the json storage.
-  // Defaults to process.cwd() + '/umzug.json';
-  path: process.cwd() + '/db/sequelize-meta.json'
-}
-```
-
-### SequelizeStorage
-Using the [SequelizeStorage](src/storages/SequelizeStorage.js) will create a table in your SQL database called `SequelizeMeta` containing an entry for each executed migration. You will have to pass a configured instance of Sequelize or an existing Sequelize model. Optionally you can specify the model name, table name, or column name. All major Sequelize versions are supported.
-
-#### Options
-
-```js
-{
-  // The configured instance of Sequelize.
-  // Optional if `model` is passed.
-  sequelize: instance,
+`index.js`:
 
-  // The to be used Sequelize model.
-  // Must have column name matching `columnName` option
-  // Optional if `sequelize` is passed.
-  model: model,
+```javascript
 
-  // The name of the to be used model.
-  // Defaults to 'SequelizeMeta'
-  modelName: 'Schema',
+const Sequelize = require('sequelize')
+const path = require('path')
+const Umzug = require('umzug')
 
-  // The name of table to create if `model` option is not supplied
-  // Defaults to `modelName`
-  tableName: 'Schema',
+// creates a basic sqlite database
+const sequelize = new Sequelize({
+  dialect: 'sqlite',
+  storage: './db.sqlite'
+})
 
-  // The name of table column holding migration name.
-  // Defaults to 'name'.
-  columnName: 'migration',
+const umzug = new Umzug({
+  migrations: {
+    // indicates the folder containing the migration .js files
+    path: path.join(__dirname, './migrations'),
+    // inject sequelize's QueryInterface in the migrations
+    params: [
+      sequelize.getQueryInterface()
+    ]
+  },
+  // indicates that the migration data should be store in the database
+  // itself through sequelize. The default configuration creates a table
+  // named `SequelizeMeta`.
+  storage: 'sequelize',
+  storageOptions: {
+    sequelize: sequelize
+  }
+})
 
-  // The type of the column holding migration name.
-  // Defaults to `Sequelize.STRING`
-  columnType: new Sequelize.STRING(100)
-}
+;(async () => {
```

Comment:
unnecessary semi

Reply:
Actually no, it is strictly mandatory. You can even test it to see it won't compile.

That's just a common pattern found in no-semicolon javascript coding style to force a proper understanding by the compiler.

Reply:
Right, codebase is not using semi